core/src/main/java/org/apache/iceberg/rest/TableScanResponseParser.java

```diff
@@ -105,11 +105,11 @@ public static void serializeScanTasks(
 
     if (fileScanTasks != null) {
       gen.writeArrayFieldStart(FILE_SCAN_TASKS);
-      Set<Integer> deleteFileReferences = Sets.newHashSet();
       for (FileScanTask fileScanTask : fileScanTasks) {
+        Set<Integer> deleteFileReferences = Sets.newHashSet();
```

Comment:
Agree with this fix, though the worst case is that we'd have more references than really needed for a given task right? It's not as though the wrong delete files would be resolved if I understand right.